threadpoolctl.py

```diff
@@ -1100,6 +1102,37 @@ def _find_libraries_with_enum_process_module_ex(self):
         finally:
             kernel_32.CloseHandle(h_process)
 
+    def _find_libraries_emscripten(self):
+        """Emscripten specific implementation for finding libraries.
+
+        Adapted from Pyodide CPython patch for ctypes.util.find_library. For more details, see:
+        https://github.com/pyodide/pyodide/blob/df8aa663b01c2072b9d2a85c18f9a8f13f70576f/cpython/patches/0003-Add-emscripten-platform-support-to-ctypes.util.find_.patch  # noqa: E501
+        """
+
+        def _is_wasm(filename):
+            # Return True if the given file is an WASM module
+            wasm_header = b"\x00asm"
+            with open(filename, "br") as thefile:
+                return thefile.read(4) == wasm_header
+
+        libraries = []
+        ld_library_paths = [
+            path
+            for path in os.environ.get("LD_LIBRARY_PATH", "").split(":")
+            if os.path.exists(path)
+        ]
+        for dir in ld_library_paths:
+            dir_full_paths = [os.path.join(dir, name) for name in os.listdir(dir)]
+            dir_libraries = [
+                path
+                for path in dir_full_paths
+                if path.endswith(".so") or path.endswith(".wasm") and _is_wasm(path)
+            ]
+            libraries += dir_libraries
+
+        for library in libraries:
+            self._make_controller_from_path(library)
```

Comment:
Could you paste the output of this ? I'd be interested in seing all the libs it finds this way
```suggestion
        for library in libraries:
            print(library)
            self._make_controller_from_path(library)
```